packages/launch/src/focused-launch/summary/index.tsx

```diff
@@ -327,13 +338,9 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
 		? [ defaultPaidPlan, nonDefaultPaidPlan, defaultFreePlan ]
 		: [ defaultPaidPlan, defaultFreePlan ];
 
-	const allAvailablePlansProducts = useSelect(
-		( select ) =>
-			allAvailablePlans.map( ( plan ) =>
-				select( PLANS_STORE ).getPlanProduct( plan?.periodAgnosticSlug, billingPeriod )
-			),
-		[ allAvailablePlans, billingPeriod ]
-	);
+	const allAvailablePlansProducts: ( PlanProduct | undefined )[] = nonDefaultPaidPlanProduct
+		? [ defaultPaidPlanProduct, nonDefaultPaidPlanProduct, defaultFreePlanProduct ]
+		: [ defaultPaidPlanProduct, defaultFreePlanProduct ];
```

Comment:
I think they will never be out of sync. Two factors determine their order:

1. [These two](https://github.com/Automattic/wp-calypso/blob/bc18130ef55f6063d508160b636808a036590e1f/packages/launch/src/focused-launch/summary/index.tsx#L321-L322) synchronous lines. Because they control the ternary.
2. [These lines that use the same resolver](https://github.com/Automattic/wp-calypso/blob/970d6d6d049c8e8babb740edae0393e401268e22/packages/launch/src/hooks/use-plans.tsx#L32-L41) that won't resolve until all is fetched.

Either ways, I cleaned up that code a bit.